docs: update Getting started docs (#47)
filecoin-station/core: README.md

	<br>
</h1>

> Station Core is a headless version of Filecoin Station suitable for running on
> all kinds of servers.

[![CI](https://github.com/filecoin-station/core/actions/workflows/ci.yml/badge.svg)](https://github.com/filecoin-station/core/actions/workflows/ci.yml)

## Deployment

Station Core supports different deployment options:

- [Docker](#docker)
- [Manual Deployment (Ubuntu)](#manual-deployment-ubuntu)

## Installation

> Station Core requires Node.js version 18.12 or newer. We recommend using the

Reviewer: see Node.js version discussion

Author: Resolved in https://github.com/filecoin-station/core/pull/47/commits/5cb8a745a1c020e995bfffc2b7c12acc118f806b

	<br>
</h1>

> Station Core is a headless version of Filecoin Station suitable for running on
> all kinds of servers.

[![CI](https://github.com/filecoin-station/core/actions/workflows/ci.yml/badge.svg)](https://github.com/filecoin-station/core/actions/workflows/ci.yml)

## Deployment

Station Core supports different deployment options:

- [Docker](#docker)
- [Manual Deployment (Ubuntu)](#manual-deployment-ubuntu)

## Installation

> **Note**: Station Core requires Node.js, we recommend using the latest LTS
> version. You can install Node.js using your favorite package manager or get
> the official installer from

> > [Node.js downloads](https://nodejs.org/en/download/).

Author: Rendered version:

<img width="1045" alt="Screenshot 2023-03-15 at 14 36 13" src="https://user-images.githubusercontent.com/1140553/225325014-82d41a01-f9f1-4db2-8f22-9221cd75b772.png">

See https://github.com/community/community/discussions/16925

Reviewer: This is looking awesome! 👏

	<br>
</h1>

> Station Core is a headless version of Filecoin Station suitable for running on
> all kinds of servers.

[![CI](https://github.com/filecoin-station/core/actions/workflows/ci.yml/badge.svg)](https://github.com/filecoin-station/core/actions/workflows/ci.yml)

## Deployment

Station Core supports different deployment options:

- [Docker](#docker)
- [Manual Deployment (Ubuntu)](#manual-deployment-ubuntu)

## Installation

Station Core requires Node.js, it supports all maintained LTS major versions. We
recommend using the latest LTS version.

```bash

> $ npm install -g --engine-strict @filecoin-station/core

Reviewer: Meh, that's ugly to me :| It should be the user's choice how they want to use NPM.

> $ npm install -g --engine-strict @filecoin-station/core

Author: 👍🏻

$ FIL_WALLET_ADDRESS=f1... station
```

## Common Configuration

Station Core is configured using environment variables (see
[The Twelve-Factor App](https://12factor.net/config)).

The following configuration options are shared by all Station commands:

- `$XDG_STATE_HOME` _(string; optional)_: Station stores logs and module state
  in the directory `$XDG_STATE_HOME/filecoin-station`. State home defaults to
  `~/.local/state`.

## Commands

### `$ station`

Start a new Station process. The Station will run in foreground and can be
terminated by pressing Ctrl+C.

This command has the following additional configuration in addition to common
the configuration options described in
[Common Configuration](#common-configuration):

- `FIL_WALLET_ADDRESS` _(string; required)_: Address of the Filecoin wallet that
  will receive rewards. The value must be a mainnet address starting with `f1`.

  If you just want to give `core` a quick spin, you can use the address
  `f1abjxfbp274xpdqcpuaykwkfb43omjotacm2p3za`. Please note that any earnings
  sent there will be lost.

### `$ station metrics`

> _This command reads information from the state files created by the main Station

Reviewer: "main Station process" is a term not yet established. Also I don't think we need to talk about files here, does the user care where that data is coming from? To me this is too much an implementation detail. Therefore I wonder what you're trying to achieve by adding this text 🤔

Author: I am so glad you started this discussion, @juliangruber 🙏🏻

Reviewer: Hmm, if you compare this to `docker`, the `logs` command doesn't care if the container is running or not. I agree though that we should prevent concurrent `$ station` calls!

To me, annotating that logs are outdated doesn't make sense. The logs won't be outdated, they will be as uptodate as they can be. Only nothing is running that can produce new logs.

Can you maybe think of another CLI that works the way you propose?

Reviewer: > Maybe we can also improve all commands to print information about the location of state files at startup to make troubleshooting easier.

Are you sure this is a problem? Have you run into this?

Reviewer: > If you don't have the main Station Core process running, there will be either no metrics and no logs, or the data will be outdated.

To me it's obvious that there will be no new metrics or logs if the thing isn't running. But maybe I have a blind spot? It's the same with systemd: You can always get the logs. If you don't get new logs, you check if the service is running, and otherwise start it.

### `$ station logs <module>`

_This command reads information from the state files created by the main Station
process._

> Get a combined log from the Station Core and all Station Modules:

Reviewer: ```suggestion
Get combined logs from Station Core and all Station Modules:
```
